test/e2e/proxy_test.go

```diff
@@ -146,9 +197,38 @@ func TestProxyLoadBalancerServiceIPv6(t *testing.T) {
 	testProxyLoadBalancerService(t, true)
 }
 
+func loadBalancerClusterTestCases(t *testing.T, data *TestData, clusterUrl string, nodes, pods, podOSes []string) {
+	t.Run("Client:Node", func(t *testing.T) {
+		testLoadBalancerClusterFromNode(t, data, nodes, clusterUrl)
+	})
+	t.Run("Client:Pod", func(t *testing.T) {
+		testLoadBalancerClusterFromPod(t, data, pods, podOSes, clusterUrl)
+	})
+}
+
+func loadBalancerLocalTestCases(t *testing.T, data *TestData, localUrl string, nodes, nodeIPs, pods, podIPs, podOSes []string, hostNetwork bool) {
+	expectedClientIPs := make([]string, len(podIPs))
+	copy(expectedClientIPs, podIPs)
+	hostNetworkStr := expectedNonHostNetworkResult
+	if hostNetwork {
+		hostNetworkStr = expectedHostNetworkResult
+		for idx := range podOSes {
+			if podOSes[idx] == "windows" {
+				expectedClientIPs[idx] = nodeIPs[idx]
```

Comment:
Why reset the expected Client IP to nodeIP when it's Windows?

Reply:
I think it is related to the NAT on host. I added comment here: "There's a NetNat on Windows host doing SNAT. So if endpoint is a hostNetwork Pod request packet will be SNATed to Node IP."